### PR TITLE
Ca dw/swd calculations

### PR DESCRIPTION
**Pull request by mrchrisadams**
This is a sanity check PR to work out why the numbers in #85 seem to be coming up strange results.

### CODE REVIEW

**Comment by mrchrisadams** [comment]
If it helps, you can spin up a workspace with gitpod to play with these numbers by following the link below.

You need an account with gitpod, but that's free to set up.

https://gitpod.io/#https://github.com/thegreenwebfoundation/co2.js/pull/87

**Comment by mrchrisadams** [comment]
OK, I've switched the constants around so that the first time load uses the 0.75 figure, and the subsequent loads use the 0.25 figure.

As I understand it, this ought to match the approach documented on the SWD page too, and should mean that the number per page ought to be 0.57 using the v9 model.

The v8 version still gives the lower figure of 0.2 - this might explain why you picked up the difference.

File: src/sustainable-web-design.js [hunk 1]
```diff
@@ -169,13 +169,28 @@ class SustainableWebDesign {
     const energyBycomponent = this.energyPerByteByComponent(bytes);
     const cacheAdjustedSegmentEnergy = {};
 
+    console.log({ energyBycomponent });
```
Comment (drydenwilliams) [inline src/sustainable-web-design.js:172]: Remove this

File: src/sustainable-web-design.js [hunk 2]
```diff
@@ -169,13 +169,28 @@ class SustainableWebDesign {
...
+
+    // energyBycomponent doesn't apply any caching logic, to should be the
+    // same number as the total in the v8
+    console.log({ v9recombinedNoCaching });
+
+    // for this, we want
     for (const [key, value] of Object.entries(energyBycomponent)) {
       // represent the first load
-      cacheAdjustedSegmentEnergy[key] = value * firstView;
+      cacheAdjustedSegmentEnergy[`${key} - first`] = value * firstView;
```
Comment (drydenwilliams) [inline src/sustainable-web-design.js:187]: Not sure about this approach.

File: src/sustainable-web-design.js [hunk 3]
```diff
@@ -169,13 +169,28 @@ class SustainableWebDesign {
...
 
       // then represent the subsequent load
-      cacheAdjustedSegmentEnergy[key] += value * returnView * dataReloadRatio;
+      cacheAdjustedSegmentEnergy[`${key} - subsequent`] =
+        value * returnView * dataReloadRatio;
     }
+    console.log({ cacheAdjustedSegmentEnergy });
```
Comment (drydenwilliams) [inline src/sustainable-web-design.js:193]: Remove `console.log`

File: src/sustainable-web-design.js [hunk 4]
```diff
@@ -189,12 +204,61 @@ class SustainableWebDesign {
...
+    return v9firstVisits + v9subsequentVisits;
+  }
+
+  /*
+   * JUST FOR TEST PURPOSES
+   * Testing v0.8.0 => v0.9.0 versions
+   *
+   *
+   */
+  energyPerVisitV8(bytes) {
```
Comment (drydenwilliams) [inline src/sustainable-web-design.js:239]: Remove this, just for testing I guess?

File: src/sustainable-web-design.test.js [hunk 5]
```diff
@@ -35,20 +35,34 @@ describe("sustainable web design model", () => {
 
     it("should calculate the correct energy", () => {
       expect(swd.energyPerVisit(averageWebsiteInBytes)).toBe(
-        0.0004513362121582032
+        0.001285882415771485
       );
     });
+
+    it("should match the old calculations", () => {
```
Comment (drydenwilliams) [inline src/sustainable-web-design.test.js:42]: Can remove this.

File: src/sustainable-web-design.js [hunk 6]
```diff
@@ -189,12 +204,61 @@ class SustainableWebDesign {
    */
   energyPerVisit(bytes) {
```
Comment (drydenwilliams) [inline src/sustainable-web-design.js:205]: This function feels far away from the SWD approach on the site. Harder to understand breaking it down into each component than putting it back together. Maybe it's ok. Just for new devs (and me) I constantly think 

> "Oh, this is different, from SWD, why is this?" 

and takes a lot to understand why.

**Comment by drydenwilliams** [comment]
Just thought we can improve over time, so let's get this merged in @mrchrisadams

**Comment by mrchrisadams** [comment]
OK @drydenwilliams  - agreed. I'll merge this in, so we can get a new release, and then a make an issue to refactor this.

**Pull request merged**